std/math.zig

```diff
@@ -806,3 +806,41 @@ test "max value type" {
     const x: u32 = maxInt(i32);
     testing.expect(x == 2147483647);
 }
+
+pub fn wideMul(comptime T: type, a: T, b: T) @IntType(T.is_signed, T.bit_count * 2) {
+    const ResultInt = @IntType(T.is_signed, T.bit_count * 2);
+    return ResultInt(a) * ResultInt(b);
+}
+
+pub extern fn __muloti4(a: i128, b: i128, overflow: *c_int) i128;
+
+pub fn mulWithOverflow(comptime T: type, a: T, b: T, result: *T) bool {
+    switch (T) {
+    // TODO how to detect if this helper is not needed
+    i128 => {
```

Comment:
This special case as well as the `extern fn __muloti4` above to me are evidence that `@mulWithOverflow` should remain. The addition of `wideMul` is definitely welcome.

Reply:
You are right. I forgot about all the weird integer types zig has. This way doesn't work about u64/i64 except for that exception, and LLVM knows all the special cases on these larger integers.